File: elixir/test/solutions/0347_top_k_frequent_elements/top_k_frequent_elements_test.exs
defmodule LeetCode.Solutions.TopKFrequentElementsTest do
  use ExUnit.Case, async: true

  alias LeetCode.Solutions.TopKFrequentElements

  test "Case 1 works" do
    assert TopKFrequentElements.call([1, 1, 1, 2, 2, 3], 2) == [1, 2]
  end

  test "Case 2 works" do
    assert TopKFrequentElements.call([1], 1) == [1]
  end
end

File: ruby/test/solutions/0347_top_k_frequent_elements/top_k_frequent_elements_test.rb
# frozen_string_literal: true

require 'test_helper'
require_relative '../../../lib/solutions/0347_top_k_frequent_elements/top_k_frequent_elements'

class TopKFrequentElementsTest < Minitest::Test
  def test_case_1_works
    assert_equal [1, 2], top_k_frequent([1, 1, 1, 2, 2, 3], 2)
  end

  def test_case_2_works
    assert_equal [1], top_k_frequent([1], 1)
  end
end
